lib/puma/server.rb

```diff
@@ -224,8 +224,11 @@ def run(background=true)
             @reactor.add client
           end
         end
+
+        process_now
```

Comment:
This return value of the `ThreadPool` block (`process_now` is true when a complete request was processed by the block) is used to determine if an OOB hook should be scheduled to fire once the thread is eventually idle.